packages/auth/src/core/auth/auth_impl.ts

```diff
@@ -645,6 +647,20 @@ export class AuthImpl implements AuthInternal, _FirebaseService {
     if (heartbeatsHeader) {
       headers[HttpHeader.X_FIREBASE_CLIENT] = heartbeatsHeader;
     }
+
+    // If the App Check service exists, add the App Check token in the headers
+    const appCheckTokenResult = await this.appCheckServiceProvider
+      .getImmediate({ optional: true })
+      ?.getToken();
+    // TODO: What do we want to do if there is an error getting the token?
```

Comment:
Copied the comment from `storage/src/service.ts` but still applicable here - do we want to do any sort of error handling on the client side aside from letting any subsequent call with the dummy token fall through?